cnd/src/network/comit_ln/swaps.rs

```diff
@@ -0,0 +1,602 @@
+use crate::{
+    network::protocols::announce::{protocol::ReplySubstream, SwapDigest},
+    swap_protocols::{HanEtherereumHalightBitcoinCreateSwapParams, LocalSwapId, SharedSwapId},
+    timestamp::Timestamp,
+};
+use digest::Digest;
+use libp2p::{swarm::NegotiatedSubstream, PeerId};
+use std::collections::HashMap;
+
+#[derive(Display, thiserror::Error, Clone, Copy, Debug)]
```

Comment:
Deriving `Display` is not going to give you any helpful error. Please use `#[error]` instead.

You might also want to think about declaring single error structs and just use `anyhow::Error` for all APIs. Saves you from declaring this enum!